plugins/modules/ndo_node_profile.py

```diff
@@ -0,0 +1,338 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2025, Samita Bhattacharjee (@samiib) <samitab@cisco.com>
+
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    "metadata_version": "1.1",
+    "status": ["preview"],
+    "supported_by": "community",
+}
+
+DOCUMENTATION = r"""
+---
+module: ndo_node_profile
+short_description: Manage Node Profiles on Cisco Nexus Dashboard Orchestrator (NDO).
+description:
+- Manage Node Profiles on Cisco Nexus Dashboard Orchestrator (NDO).
+- This module is only supported on ND v3.1 (NDO v4.3) and later.
+author:
+- Samita Bhattacharjee (@samiib)
+options:
+  template:
+    description:
+    - The name of the template.
+    - The template must be a Fabric Resource Policy template.
+    - This parameter or O(template_id) is required.
+    type: str
+  template_id:
+    description:
+    - The ID of the template.
+    - The template must be a Fabric Resource Policy template.
+    - This parameter or O(template) is required.
+    type: str
+  name:
+    description:
+    - The name of the Node Profile.
+    type: str
+    aliases: [ node_profile ]
+  uuid:
+    description:
+    - The UUID of the Node Profile.
+    - This parameter is required when the O(name) needs to be updated.
+    type: str
+    aliases: [ node_profile_uuid ]
+  description:
+    description:
+    - The description of the Node Profile.
+    type: str
+  nodes:
+    description:
+    - The list of node IDs to associate with the Node Profile.
+    - Each element can either be a single node ID or a range of IDs in quotation marks ('').
```

Comment:
Sorry, looks like ('') is not required here. So reverting my previous comment.

- Each element can either be a single node ID or a range of IDs.